### ISSUE TITLE
Stepping into sources does not work with `vcpkg_install_msbuild` (removes `-dbg`)

### ISSUE THREAD

**Issue by HofmeisterAn** [microsoft/vcpkg#23547]
**Describe the bug**
Stepping into packages built with `vcpkg_install_msbuild` does not work. The sources referenced by the `pdb` file are in `buildtrees\<port>\<arch>-windows-dbg`, but `vcpkg_clean_msbuild` removes this directory:
https://github.com/microsoft/vcpkg/blob/965961552d9dca07ee88e89c8a236afdda2d7a14/scripts/cmake/vcpkg_clean_msbuild.cmake#L21

**Environment**
- OS: Windows
- Compiler: MSVC 14.29.30133

**To Reproduce**
1. Consume a port that uses `vcpkg_install_msbuild`
2. Try to step into the sources in Visual Studio

**Expected behavior**
A developer can step into the sources

**Failure logs** 
\-

**Additional context**
- It looks like `vcpkg_configure_cmake` does not remove the debug directory. At least it exists in `buildtrees\<port>\`
- [SKIP_CLEAN](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_install_msbuild.md#skip_clean) "fixes" the issue, but keeps `-rel` too

**Comment by MonicaLiu0311** [comment]
>[SKIP_CLEAN](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_install_msbuild.md#skip_clean) "fixes" the issue, but keeps -rel too

Yes, as you said, this is a `by design`, `vcpkg_install_msbuild` cleans the intermediate files by default, if you want to keep, please use [SKIP_CLEAN](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_install_msbuild.md#skip_clean), then neither `${TARGET_TRIPLET}-rel` nor `${TARGET_TRIPLET}-dbg` will be clean up.

**Comment by MonicaLiu0311** [comment]
We hope your question was answered to your satisfaction; if it wasn't, you can reopen with more info.

**Comment by HofmeisterAn** [comment]
> Yes, as you said, this is a `by design`

Well — I never said that, odd response.

> `vcpkg_install_msbuild` cleans the intermediate files by default

That makes debugging hard. Considering `vcpkg_configure_cmake` does handle it right, I still think it is a bug.

**Comment by MonicaLiu0311** [comment]
>>>[SKIP_CLEAN](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_install_msbuild.md#skip_clean) "fixes" the issue, but keeps -rel too
>>
> > Yes, as you said, this is a `by design`
> 
> Well — I never said that, odd response.

Sorry for the ambiguity, re-answering the question. Fact as you said, SKIP_CLEAN will preserve `${TARGET_TRIPLET}-rel` and `${TARGET_TRIPLET}-dbg` , but this is a `by design`, not a bug. 

Based on your comments, this issue will be reopened and converted to discussion.
